.github/workflows/package.yaml

```diff
@@ -43,8 +42,129 @@ jobs:
           python setup.py bdist_wheel
           cp dist/*.whl docker/root
 
+      - name: Save Python build cache
+        uses: actions/cache@v3
+        with:
+          path: docker/root
+          key: ${{github.sha}}
+
+
+  # Build ARM64 container
+  package-arm64:
+    runs-on: ubuntu-latest
+    needs: [
+      build
+    ]
+
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Restore Python build cache
+        uses: actions/cache@v3
+        with:
+          path: docker/root
+          key: ${{github.sha}}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: linux/arm64
+
+      - name: Docker Setup Buildx
+        uses: docker/setup-buildx-action@v2.0.0
+
+      - name: Build Docker Image
+        run: |
+          docker buildx build \
+          --platform=linux/arm64 \
+          --cache-to=type=local,dest=/tmp/build-cache/arm64 \
+          --tag $IMAGE_NAME \
+          --label "runnumber=${GITHUB_RUN_ID}" \
+          docker/
+
+      - name: Save ARM64 build cache
+        uses: actions/cache@v3
+        with:
+          path: /tmp/build-cache/arm64
+          key: ${{github.sha}}
+
+
+  # Build AMD64 container
+  package-amd64:
+    runs-on: ubuntu-latest
+    needs: [
+      build
+    ]
+
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Restore Python build cache
+        uses: actions/cache@v3
+        with:
+          path: docker/root
+          key: ${{github.sha}}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: linux/amd64
+
+      - name: Docker Setup Buildx
+        uses: docker/setup-buildx-action@v2.0.0
+
       - name: Build Docker Image
-        run: docker build --tag $IMAGE_NAME --label "runnumber=${GITHUB_RUN_ID}" docker/
+        run: |
+          docker buildx build \
+          --platform=linux/amd64 \
+          --cache-to=type=local,dest=/tmp/build-cache/amd64 \
+          --tag $IMAGE_NAME \
+          --label "runnumber=${GITHUB_RUN_ID}" \
+          docker/
+
+      - name: Save AMD64 build cache
+        uses: actions/cache@v3
+        with:
+          path: /tmp/build-cache/amd64
+          key: ${{github.sha}}
+
+
+  # On master branch, build the docker manifest file from the cached docker builds and push to the registry
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [
+      build,
+      package-arm64,
+      package-amd64
+    ]
+
+    permissions:
+      packages: write
+      contents: read
+
+    if: contains('refs/heads/master', github.ref) || ${{ github.ref }} =~ ^refs\/tags\/v.*
```

Comment:
Need to add `(...) && ${GITHUB_REPOSITORY} == 'jmbannon\/ytdl-sub' here so it it skips this at PR time. There might be a more elegant way to do this

Reply:
It's still running because in your fork, the branch name is master

Reply:
I don't think we should limit it to the repository then forks wouldn't build anymore. This build is using "refs/remotes/pull/84/merge" so it shouldn't have picked up this final deploy job. Let me check if merge can be ignored.

Reply:
Good point. I think we can use `(...) && ${{ github.event_name == 'push' }}`

Reply:
That was my thought as well, but it seemed to be some formatting on the if statement causing it to run too often. Pushed an update to the PR for this.